tests/unittests/tests-libc/tests-libc.c

```diff
@@ -30,6 +30,31 @@ static void test_libc_strscpy(void)
     TEST_ASSERT_EQUAL_INT(strscpy(buffer, "empty", 0), -E2BIG);
 }
 
+static void test_libc_swprintf(void)
+{
+    string_writer_t sw;
+    char buffer[32];
+    int res;
+
+    string_writer_init(&sw, buffer, sizeof(buffer));
+
+    res = swprintf(&sw, "Hello World!");
+    TEST_ASSERT_EQUAL_INT(res, 12);
+    TEST_ASSERT_EQUAL_INT(string_writer_len(&sw), 12);
+    TEST_ASSERT_EQUAL_INT(strcmp("Hello World!", string_writer_str(&sw)), 0);
+
+    res = swprintf(&sw, "0123456789");
+    TEST_ASSERT_EQUAL_INT(res, 10);
+    res = swprintf(&sw, "01234567891");
+    TEST_ASSERT_EQUAL_INT(res, -E2BIG);
+    res = swprintf(&sw, "###");
+    TEST_ASSERT_EQUAL_INT(res, -E2BIG);
```

Comment:
I think it is worth it, to use a new `string_writer_t` + buffer. Not because it's technically needed but it signals to the reader "we are testing a new/different behavior". Also, which behavior is tested here? ;)

Reply:
No this is all the same string write - we want to append to it until it's full and test at every step that it behaves as expected.

I added comments to better explain what's going on.